ansible_builder/main.py

```diff
@@ -228,43 +245,41 @@ def prepare_appended_steps(self):
     def prepare_introspection_steps(self):
         source = ansible_builder.introspect.__file__
         dest = os.path.join(self.build_context, 'introspect.py')
-        exists = os.path.exists(dest)
-        if not exists or not filecmp.cmp(source, dest, shallow=False):
-            shutil.copy(source, dest)
+        copy_file(source, dest)
+
+        user_files = {'python': None, 'system': None}
+        for item, context_name in CONTEXT_FILES.items():
+            if self.definition.get_dep_abs_path(item):
+                user_files[item] = context_name
 
-        self.steps.extend(IntrospectionSteps(os.path.basename(dest)))
+        self.steps.extend(
+            IntrospectionSteps(
+                'introspect.py', user_files['python'], user_files['system'],
+                BINDEP_COMBINED
+            )
+        )
 
     def prepare_galaxy_steps(self):
-        galaxy_file = self.definition.get_dep('galaxy')
-        if galaxy_file:
-            self.steps.extend(GalaxySteps(galaxy_file))
+        if self.definition.get_dep_abs_path('galaxy'):
+            self.steps.extend(GalaxySteps(CONTEXT_FILES['galaxy']))
         return self.steps
 
-    def prepare_pip_steps(self):
-        python_req_file = self.definition.get_dep('python')
-
-        command = [self.container_runtime, "run", "--rm", self.tag, "introspect"]
-        rc, output = run_command(command, capture_output=True)
-        data = yaml.safe_load("\n".join(output))
+    def prepare_pip_steps(self, collection_pip):
+        pip_list = sanitize_requirements(collection_pip)
 
-        if python_req_file:
-            with open(self.definition.get_dep_abs_path('python'), 'r') as f:
-                user_py_reqs = f.read().split('\n')
-            data['python'].extend(user_py_reqs)
-        data['python'] = sanitize_requirements(data['python'])
-        pip_file = os.path.join(self.build_context, 'requirements.txt')
```

Comment:
Combining user's `requirements.txt` file was moved earlier in the process (phase 1 as opposed to phase 2), so that's why it's axed here. This was a phase 2 thing.